junit-jupiter-api/src/main/java/org/junit/jupiter/api/Assertions.java

```diff
@@ -54,6 +54,9 @@ private Assertions() {
 	 * <p>See Javadoc for {@link #fail(String, Throwable)} for an explanation of
 	 * this method's generic return type {@code V}.
 	 */
+	// we purposely return a V that has nothing to do with the parameters of this method; see
+	// the javadoc for details.
+	@SuppressWarnings("TypeParameterUnusedInFormals")
```

Comment:
It is indeed non-standard. It's an error-prone warning.

I added it here because I thought that there weren't enough false positives to justify disabling the corresponding error-prone rule "TypeParameterUnusedInFormals" globally. If you disagree, I'd be happy to disable "TypeParameterUnusedInFormals" by adding ""-Xep:TypeParameterUnusedInFormals:OFF" to the `build.gradle` file.